File: core-java-practice/leet-code-codebase/ValidParentheses.java
import java.util.*;

public class ValidParentheses {
    public static boolean check(String str){
            //initializing stack for open brackets
            Stack<Character> st =new Stack<>();
            for(char ch: str.toCharArray()){
                //storing the open brackets
                if(ch=='[' || ch=='{' || ch=='('){
                    st.push(ch);
                }
                else{
                    //if already empty stack so false
                    if(st.isEmpty()){
                        return false;
                    }
                    //top character in stack
                    char top=st.pop();
                    //checking all the edge cases
                    if(ch==')' && top!='('){
                        return false;
                    }
                    if(ch==']' && top!='['){
                        return false;
                    }
                    if(ch=='}' && top!='{'){
                        return false;
                    }
                }
            }
            return st.isEmpty();
    }
}

File: core-csharp-practice/leet-code-codebase/ValidParentheses.cs
using System;

class ValidParetheses{
      public static bool isValid(string str){
        Stack<char> st=new Stack<>();

        foreach(char ch in str){
            //push opening brackets
            if(ch=='(' || ch=='{' || ch=='['){
                st.push(ch);
            }
            else{
                //edge case
                if(st.Count==0){
                    return false;
                }
                //top element of stack
                char top=st.Pop();
                
                if(ch==']' && top!='['){
                    return false;
                }
                if(ch==')' && top!='('){
                    return false;
                }
                if(ch=='}' && top!='{'){
                    return false;
                }
            }
        }
        return st.Count==0; //if the stack is empty string is valid
      }
}